modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java

```diff
@@ -278,4 +286,40 @@ private void enrichParameter(OfValues resolverClass,
     }
     return parameterNames;
   }
+
+  private Optional<?> getOfValueAnnotation(ParameterDeclaration parameterDeclaration) {
```

Comment:
this goes against the pattern that we have defined and used across all other FC features. We want one single code path capable of handling both cases. If an SDK-API implementation is found, then use it. If a legacy implementation is used, then use an adapter wrapper so that it can be treated as as if it were an sdk-api one. `LegacyResultAdapter` and `LegacyPollingSourceAdapter` are examples of this pattern. 

Using wildcard and `Either` classes are not a scalable pattern

Reply:
How can we use an adapter with an annotation o or a class?
We are using Either right now by the way: https://github.com/mulesoft/mule/commit/8eceaac2b1422e2a1c29b1897a16eaba28bc38cc